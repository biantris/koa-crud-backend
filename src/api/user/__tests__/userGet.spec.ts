import {
  clearDbAndRestartCounters,
  connectMongoose,
  disconnectMongoose,
  createGetApiCall,
} from "../../../../test";
import { createUser } from "../../../modules/user/fixtures/createUser";

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

const getUrl = (id: string) => `/api/user/${id}`;

it("should return error if user it was not found", async () => {
  const response = await createGetApiCall({
    url: getUrl("5c42132aa591a2001ad46264"),
  });

  expect(response.status).toBe(400);
  expect(response.body.message).toBe("User not found");
  expect(response.body).toMatchSnapshot();
});

it("should return user by id", async () => {
  const user = await createUser({});
  const response = await createGetApiCall({
    url: getUrl(user._id),
  });

  expect(response.status).toBe(200);
  expect(response.body.user.name).toBe(user.name);
  expect(response.body.user.email).toBe(user.email);
  expect(response.body).toMatchSnapshot();
});
