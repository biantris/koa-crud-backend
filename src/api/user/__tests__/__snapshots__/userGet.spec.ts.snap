// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return error if user it was not found 1`] = `
Object {
  "message": "User not found",
}
`;

exports[`should return user by id 1`] = `
Object {
  "user": Object {
    "_id": "60af1420eff8ea127b5a5547",
    "email": "user0@test.com",
    "name": "User 0",
  },
}
`;
